module_utils/network_lsr/argument_validator.py

```diff
@@ -1491,6 +1496,14 @@ def _validate_post(self, value, name, result):
                         "but is '%s'" % result["mac"],
                     )
 
+            if "interface_path" in result:
+                if result["type"] not in ["ethernet", "infiniband"]:
+                    raise ValidationError(
+                        name + ".interface_path",
+                        "a 'interface_path' address is only allowed for type 'ethernet' "
+                        "or 'infiniband'",
+                    )
```

Comment:
with above comment about not using `default_value=list`, you may want to treat an empty list here somehow special.

but in practice not. All the choices that you have here (like rejecting an empty list, or normalizing it to `None`) seem not desirable. So, it's fine that the ArgValidator accepts both a missing and empty option here (and preserves them both).

Reply:
I added the following to argvalidator to handle the situation where `interface_path` == []:
```
         if "interface_path" in result:
                if result["interface_path"] == []:
                    raise ValidationError(
                        name + ".interface_path",
                        "'interface_path' address can not be empty list",
                    )
```